packages/libs/src/utils/estimateGas.ts

```diff
@@ -17,32 +17,34 @@ export interface ContractMethod {
   encodeABI: () => string
   send: <Tx>(config: {
     from: Wallet | string | undefined
-    gas: number
+    gas?: number
     gasPrice?: number
   }) => Tx
 }
 
 interface EstimateGasConfig {
+  // The contract method
   method: ContractMethod
+  // Address the method will be sent from (required if the contract requires a certain sender, e.g. guardian)
   from?: Wallet | string
+  // The maximum amount of gas we will allow (likely will return a number much smaller than this)
   gasLimitMaximum: number
+  // The multiplier to safe-guard against estimates that are too low
   multiplier?: number
+  // Whether or not to throw if gas estimation fails. Usually this signals that a
+  // contract call's simulation failed and likely the actual contract call will fail.
+  shouldThrow?: boolean
```

Comment:
`shouldThrowIfGasEstimationFails` might be an annoyingly long but better name - `shouldThrow` makes it sound like it should always throw

Reply:
I like that!